src/nsidc/metgen/metgen.py

```diff
@@ -359,18 +358,17 @@ def process(configuration: config.Config) -> None:
     summarize_results(results)
 
 
-def data_reader(
-    auth_id: str, data_file: str
-) -> Callable[[str, str, str, config.Config], dict]:
+def data_reader(data_files: set[str]) -> Callable[[str, str, str, config.Config], dict]:
```

Comment:
See the PR for the issue 169 patch. Need to use the "reference data file" here, not an arbitrary file.